distributed-calculator/csharp/Dockerfile

```diff
@@ -1,16 +1,18 @@
-FROM mcr.microsoft.com/dotnet/core/sdk:3.1 AS build-env
+FROM mcr.microsoft.com/dotnet/sdk:5.0-buster-slim-amd64 AS build-env
```

Comment:
Build step for dotnet core must be on amd64 and the platform is set at the restore command.